src/App.tsx

```diff
@@ -17,13 +28,57 @@ function App() {
     };
 
     setTaskList([...taskList, newTask]);
+
+    setTaskText('');
+  };
+
+  const clickEdit = (id: string) => {
+    const taskToEdit: Task[] = taskList.filter((task: Task) => {
+      return task.id === id;
+    });
+
+    setIsEdit(true);
+
+    setIdEdit(id);
+
+    setTaskText(taskToEdit[0].textTask);
+  };
```

Comment:
aqui voce pode só setar o estado do id e dps na funçao de editar voce usa ele para filtrar a lista e editar somente o que tiver o id que esta no estado 

```suggestion
  const startEditing = (id: string) => {
     setEditingID(id)
     setTaskText(taskList.filter( (task) => id  === task.id).text)
  };
```